File: app/src/test/java/com/slapshotapps/dragonshockey/models/GameTest.java
package com.slapshotapps.dragonshockey.models;

import org.junit.Assert;
import org.junit.Test;

/**
 * Created on 10/21/16.
 */

public class GameTest {

    @Test
    public void testEquals_falseGameID() {
        Game originalGame = new Game();

        originalGame.setGameID(2);
        originalGame.setGameTime("hi");
        originalGame.setOpponent("predators");
        originalGame.setGameResult(new GameResult());
        originalGame.getGameResult().setDragonsScore(9);
        originalGame.getGameResult().setOpponentScore(7);
        originalGame.getGameResult().setGameID(2);
        originalGame.getGameResult().setOvertimeLoss(false);

        Game modifiedGame = new Game();
        modifiedGame.setGameID(3);
        modifiedGame.setGameTime("hi");
        modifiedGame.setOpponent("predators");
        modifiedGame.setGameResult(new GameResult());
        modifiedGame.getGameResult().setDragonsScore(9);
        modifiedGame.getGameResult().setOpponentScore(7);
        modifiedGame.getGameResult().setGameID(2);
        modifiedGame.getGameResult().setOvertimeLoss(false);

        Assert.assertFalse(originalGame.equals(modifiedGame));
    }

    @Test
    public void testEquals_falseGameTime() {
        Game originalGame = new Game();

        originalGame.setGameID(2);
        originalGame.setGameTime("hi");
        originalGame.setOpponent("predators");
        originalGame.setGameResult(new GameResult());
        originalGame.getGameResult().setDragonsScore(9);
        originalGame.getGameResult().setOpponentScore(7);
        originalGame.getGameResult().setGameID(2);
        originalGame.getGameResult().setOvertimeLoss(false);

        Game modifiedGame = new Game();
        modifiedGame.setGameID(2);
        modifiedGame.setGameTime("hi3");
        modifiedGame.setOpponent("predators");
        modifiedGame.setGameResult(new GameResult());
        modifiedGame.getGameResult().setDragonsScore(9);
        modifiedGame.getGameResult().setOpponentScore(7);
        modifiedGame.getGameResult().setGameID(2);
        modifiedGame.getGameResult().setOvertimeLoss(false);

        Assert.assertFalse(originalGame.equals(modifiedGame));
    }

    @Test
    public void testEquals_falseOpponent() {
        Game originalGame = new Game();

        originalGame.setGameID(2);
        originalGame.setGameTime("hi");
        originalGame.setOpponent("predators");
        originalGame.setGameResult(new GameResult());
        originalGame.getGameResult().setDragonsScore(9);
        originalGame.getGameResult().setOpponentScore(7);
        originalGame.getGameResult().setGameID(2);
        originalGame.getGameResult().setOvertimeLoss(false);

        Game modifiedGame = new Game();
        modifiedGame.setGameID(2);
        modifiedGame.setGameTime("hi");
        modifiedGame.setOpponent("wings");
        modifiedGame.setGameResult(new GameResult());
        modifiedGame.getGameResult().setDragonsScore(9);
        modifiedGame.getGameResult().setOpponentScore(7);
        modifiedGame.getGameResult().setGameID(2);
        modifiedGame.getGameResult().setOvertimeLoss(false);

        Assert.assertFalse(originalGame.equals(modifiedGame));
    }

    @Test
    public void testEquals_falseOTL() {
        Game originalGame = new Game();

        originalGame.setGameID(2);
        originalGame.setGameTime("hi");
        originalGame.setOpponent("predators");
        originalGame.setGameResult(new GameResult());
        originalGame.getGameResult().setDragonsScore(9);
        originalGame.getGameResult().setOpponentScore(7);
        originalGame.getGameResult().setGameID(2);
        originalGame.getGameResult().setOvertimeLoss(false);

        Game modifiedGame = new Game();
        modifiedGame.setGameID(2);
        modifiedGame.setGameTime("hi");
        modifiedGame.setOpponent("predators");
        modifiedGame.setGameResult(new GameResult());
        modifiedGame.getGameResult().setDragonsScore(9);
        modifiedGame.getGameResult().setOpponentScore(7);
        modifiedGame.getGameResult().setGameID(2);
        modifiedGame.getGameResult().setOvertimeLoss(true);

        Assert.assertFalse(originalGame.equals(modifiedGame));
    }

    @Test
    public void testEquals_falseDragons() {
        Game originalGame = new Game();

        originalGame.setGameID(2);
        originalGame.setGameTime("hi");
        originalGame.setOpponent("predators");
        originalGame.setGameResult(new GameResult());
        originalGame.getGameResult().setDragonsScore(9);
        originalGame.getGameResult().setOpponentScore(7);
        originalGame.getGameResult().setGameID(2);
        originalGame.getGameResult().setOvertimeLoss(false);

        Game modifiedGame = new Game();
        modifiedGame.setGameID(2);
        modifiedGame.setGameTime("hi");
        modifiedGame.setOpponent("predators");
        modifiedGame.setGameResult(new GameResult());
        modifiedGame.getGameResult().setDragonsScore(10);
        modifiedGame.getGameResult().setOpponentScore(7);
        modifiedGame.getGameResult().setGameID(2);
        modifiedGame.getGameResult().setOvertimeLoss(false);

        Assert.assertFalse(originalGame.equals(modifiedGame));
    }

    @Test
    public void testEquals_falseOpponentScore() {
        Game originalGame = new Game();

        originalGame.setGameID(2);
        originalGame.setGameTime("hi");
        originalGame.setOpponent("predators");
        originalGame.setGameResult(new GameResult());
        originalGame.getGameResult().setDragonsScore(9);
        originalGame.getGameResult().setOpponentScore(7);
        originalGame.getGameResult().setGameID(2);
        originalGame.getGameResult().setOvertimeLoss(false);

        Game modifiedGame = new Game();
        modifiedGame.setGameID(2);
        modifiedGame.setGameTime("hi");
        modifiedGame.setOpponent("predators");
        modifiedGame.setGameResult(new GameResult());
        modifiedGame.getGameResult().setDragonsScore(9);
        modifiedGame.getGameResult().setOpponentScore(6);
        modifiedGame.getGameResult().setGameID(2);
        modifiedGame.getGameResult().setOvertimeLoss(false);

        Assert.assertFalse(originalGame.equals(modifiedGame));
    }

    @Test
    public void testEquals_true() {
        Game originalGame = new Game();

        originalGame.setGameID(2);
        originalGame.setGameTime("hi");
        originalGame.setOpponent("predators");
        originalGame.setGameResult(new GameResult());
        originalGame.getGameResult().setDragonsScore(9);
        originalGame.getGameResult().setOpponentScore(7);
        originalGame.getGameResult().setGameID(2);
        originalGame.getGameResult().setOvertimeLoss(false);

        Game modifiedGame = new Game();
        modifiedGame.setGameID(2);
        modifiedGame.setGameTime("hi");
        modifiedGame.setOpponent("predators");
        modifiedGame.setGameResult(new GameResult());
        modifiedGame.getGameResult().setDragonsScore(9);
        modifiedGame.getGameResult().setOpponentScore(7);
        modifiedGame.getGameResult().setGameID(2);
        modifiedGame.getGameResult().setOvertimeLoss(false);

        Assert.assertTrue(originalGame.equals(modifiedGame));
    }
}

File: app/src/test/java/com/slapshotapps/dragonshockey/models/GameTest.kt
package com.slapshotapps.dragonshockey.models

import org.junit.Assert
import org.junit.Test

/**
 * Created on 10/21/16.
 */

class GameTest {

    @Test
    fun testEquals_falseGameID() {
        val originalGame = Game()

        originalGame.gameID = 2
        originalGame.gameTime = "hi"
        originalGame.opponent = "predators"
        originalGame.gameResult = GameResult()
        originalGame.gameResult!!.dragonsScore = 9
        originalGame.gameResult!!.opponentScore = 7
        originalGame.gameResult!!.gameID = 2
        originalGame.gameResult!!.overtimeLoss = false

        val modifiedGame = Game()
        modifiedGame.gameID = 3
        modifiedGame.gameTime = "hi"
        modifiedGame.opponent = "predators"
        modifiedGame.gameResult = GameResult()
        modifiedGame.gameResult!!.dragonsScore = 9
        modifiedGame.gameResult!!.opponentScore = 7
        modifiedGame.gameResult!!.gameID = 2
        modifiedGame.gameResult!!.overtimeLoss = false

        Assert.assertFalse(originalGame == modifiedGame)
    }

    @Test
    fun testEquals_falseGameTime() {
        val originalGame = Game()

        originalGame.gameID = 2
        originalGame.gameTime = "hi"
        originalGame.opponent = "predators"
        originalGame.gameResult = GameResult()
        originalGame.gameResult!!.dragonsScore = 9
        originalGame.gameResult!!.opponentScore = 7
        originalGame.gameResult!!.gameID = 2
        originalGame.gameResult!!.overtimeLoss = false

        val modifiedGame = Game()
        modifiedGame.gameID = 2
        modifiedGame.gameTime = "hi3"
        modifiedGame.opponent = "predators"
        modifiedGame.gameResult = GameResult()
        modifiedGame.gameResult!!.dragonsScore = 9
        modifiedGame.gameResult!!.opponentScore = 7
        modifiedGame.gameResult!!.gameID = 2
        modifiedGame.gameResult!!.overtimeLoss = false

        Assert.assertFalse(originalGame == modifiedGame)
    }

    @Test
    fun testEquals_falseOpponent() {
        val originalGame = Game()

        originalGame.gameID = 2
        originalGame.gameTime = "hi"
        originalGame.opponent = "predators"
        originalGame.gameResult = GameResult()
        originalGame.gameResult!!.dragonsScore = 9
        originalGame.gameResult!!.opponentScore = 7
        originalGame.gameResult!!.gameID = 2
        originalGame.gameResult!!.overtimeLoss = false

        val modifiedGame = Game()
        modifiedGame.gameID = 2
        modifiedGame.gameTime = "hi"
        modifiedGame.opponent = "wings"
        modifiedGame.gameResult = GameResult()
        modifiedGame.gameResult!!.dragonsScore = 9
        modifiedGame.gameResult!!.opponentScore = 7
        modifiedGame.gameResult!!.gameID = 2
        modifiedGame.gameResult!!.overtimeLoss = false

        Assert.assertFalse(originalGame == modifiedGame)
    }

    @Test
    fun testEquals_falseOTL() {
        val originalGame = Game()

        originalGame.gameID = 2
        originalGame.gameTime = "hi"
        originalGame.opponent = "predators"
        originalGame.gameResult = GameResult()
        originalGame.gameResult!!.dragonsScore = 9
        originalGame.gameResult!!.opponentScore = 7
        originalGame.gameResult!!.gameID = 2
        originalGame.gameResult!!.overtimeLoss = false

        val modifiedGame = Game()
        modifiedGame.gameID = 2
        modifiedGame.gameTime = "hi"
        modifiedGame.opponent = "predators"
        modifiedGame.gameResult = GameResult()
        modifiedGame.gameResult!!.dragonsScore = 9
        modifiedGame.gameResult!!.opponentScore = 7
        modifiedGame.gameResult!!.gameID = 2
        modifiedGame.gameResult!!.overtimeLoss = true

        Assert.assertFalse(originalGame == modifiedGame)
    }

    @Test
    fun testEquals_falseDragons() {
        val originalGame = Game()

        originalGame.gameID = 2
        originalGame.gameTime = "hi"
        originalGame.opponent = "predators"
        originalGame.gameResult = GameResult()
        originalGame.gameResult!!.dragonsScore = 9
        originalGame.gameResult!!.opponentScore = 7
        originalGame.gameResult!!.gameID = 2
        originalGame.gameResult!!.overtimeLoss = false

        val modifiedGame = Game()
        modifiedGame.gameID = 2
        modifiedGame.gameTime = "hi"
        modifiedGame.opponent = "predators"
        modifiedGame.gameResult = GameResult()
        modifiedGame.gameResult!!.dragonsScore = 10
        modifiedGame.gameResult!!.opponentScore = 7
        modifiedGame.gameResult!!.gameID = 2
        modifiedGame.gameResult!!.overtimeLoss = false

        Assert.assertFalse(originalGame == modifiedGame)
    }

    @Test
    fun testEquals_falseOpponentScore() {
        val originalGame = Game()

        originalGame.gameID = 2
        originalGame.gameTime = "hi"
        originalGame.opponent = "predators"
        originalGame.gameResult = GameResult()
        originalGame.gameResult!!.dragonsScore = 9
        originalGame.gameResult!!.opponentScore = 7
        originalGame.gameResult!!.gameID = 2
        originalGame.gameResult!!.overtimeLoss = false

        val modifiedGame = Game()
        modifiedGame.gameID = 2
        modifiedGame.gameTime = "hi"
        modifiedGame.opponent = "predators"
        modifiedGame.gameResult = GameResult()
        modifiedGame.gameResult!!.dragonsScore = 9
        modifiedGame.gameResult!!.opponentScore = 6
        modifiedGame.gameResult!!.gameID = 2
        modifiedGame.gameResult!!.overtimeLoss = false

        Assert.assertFalse(originalGame == modifiedGame)
    }

    @Test
    fun testEquals_true() {
        val originalGame = Game()

        originalGame.gameID = 2
        originalGame.gameTime = "hi"
        originalGame.opponent = "predators"
        originalGame.gameResult = GameResult()
        originalGame.gameResult!!.dragonsScore = 9
        originalGame.gameResult!!.opponentScore = 7
        originalGame.gameResult!!.gameID = 2
        originalGame.gameResult!!.overtimeLoss = false

        val modifiedGame = Game()
        modifiedGame.gameID = 2
        modifiedGame.gameTime = "hi"
        modifiedGame.opponent = "predators"
        modifiedGame.gameResult = GameResult()
        modifiedGame.gameResult!!.dragonsScore = 9
        modifiedGame.gameResult!!.opponentScore = 7
        modifiedGame.gameResult!!.gameID = 2
        modifiedGame.gameResult!!.overtimeLoss = false

        Assert.assertTrue(originalGame == modifiedGame)
    }
}
